apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/EndpointJsonComparatorBase.java

```diff
@@ -0,0 +1,163 @@
+package gov.cms.bfd.server.war;
+
+import static io.restassured.RestAssured.given;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import gov.cms.bfd.model.rif.entities.Beneficiary;
+import gov.cms.bfd.model.rif.samples.StaticRifResourceGroup;
+import gov.cms.bfd.server.war.utils.AssertUtils;
+import io.restassured.http.Header;
+import io.restassured.http.Headers;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/** Holds shared functionality between endpoint comparator tests in v1/v2. */
+public abstract class EndpointJsonComparatorBase extends ServerRequiredTest {
+
+  /** Test to use for an ignored field. */
+  protected static final String IGNORED_FIELD_TEXT = "IGNORED_FIELD";
+
+  /**
+   * Generates current endpoint response files, comparing them to the corresponding approved
+   * responses.
+   *
+   * <p>If you've recently made mapping changes, and this test fails, you may need to re-generate
+   * the expected response files by building the project with the maven parameter
+   * -DgenerateTestData=true which will run the test that generates new response files. This should
+   * only be done if you're sure your mapping changes are correct, as these files are used for
+   * regression testing future changes. Once generated, these newly minted files should be checked
+   * in with your mapping changes.
+   *
+   * @param endpointId the endpoint id
+   * @param endpointOperation the endpoint operation
+   */
+  @ParameterizedTest(name = "endpointId = {0}")
+  @MethodSource("data")
+  public void verifyCorrectEndpointResponse(String endpointId, Supplier<String> endpointOperation) {
+    String endpointResponse = endpointOperation.get();
+    assertJsonDiffIsEmpty(endpointId, endpointResponse);
```

Comment:
Test method moved into the base class, so both can share it.

Here's an improvement; this used to call the server, write the response out to a file, then immediately read the file back into a string and compare that against the expected result which was also read from a file. I've removed the superfluous file writing of the response under test, which should incrementally increase the suite speed.